lib/plugins/aws/package/compile/events/apiGateway/lib/requestValidator.js

```diff
@@ -0,0 +1,216 @@
+'use strict';
+
+const _ = require('lodash');
+const BbPromise = require('bluebird');
+
+module.exports = {
+  compileRequestValidators() {
+    const apiGateway = this.serverless.service.provider.apiGateway || {};
+    const models = {};
+    // Models are defined in the provider and can be used in http events later
+    if (apiGateway.requestSchemas) {
+      for (const [schemaId, schemaConfig] of _.entries(apiGateway.requestSchemas)) {
+        let modelName;
+        let validatorName;
+        let description;
+        let definition;
+
+        // If schema is not defined this will try to map resourceDefinition as the schema
+        if (!schemaConfig.schema) {
+          definition = schemaConfig;
+        } else {
+          definition = schemaConfig.schema;
+        }
+
+        const resourceName = _.upperFirst(_.camelCase(schemaId));
+        const modelLogicalId = this.provider.naming.getModelLogicalId(
+          resourceName,
+          'any',
+          'application/json'
+        );
+
+        const validatorLogicalId = this.provider.naming.getValidatorLogicalId(resourceName, 'any');
```

Comment:
I light of above comment I would refactor `getValidatorLogicalId` to just take `modelLogicalId`

Reply:
This one is going to be a little more involved. The resource only defines one validator as a whole and multiple models can be defined. With that route multiple validators will be created while only one is being referenced in the Method Resource.

Reply:
As i checked this method is only used in two places.. It can simply be updated to accept just `modelLogicalId` and places were it's invoked should then be updated to just pass it (and over there it's resolved upfront)

Reply:
This was updated and refactored out to another method